azurelinuxagent/common/event.py

```diff
@@ -128,9 +139,45 @@ def _save(self):
     ]
 
 
+def _encode_message(op, message):
+    """
+    Gzip and base64 encode a message based on the operation.
+
+    The intent of this message is to make the logs human readable and include the
+    stdout/stderr from extension operations.  Extension operations tend to generate
+    a lot of noise, which makes it difficult to parse the line-oriented waagent.log.
+    The compromise is to encode the stdout/stderr so we preserve the data and do
+    not destroy the line oriented nature.
+
+    The data can be recovered using the following command:
+
+      $ echo '<encoded data>' | base64 -d | pigz -zd
+
+    You may need to install the pigz command.
+
+    :param op: Operation, e.g. Enable or Install
+    :param message: Message to encode
+    :return: gzip'ed and base64 encoded message, or the original message
+    """
+
+    if len(message) == 0:
+        return message
+
+    if op not in SHOULD_ENCODE_MESSAGE_OP:
+        return message
+
+    try:
+        return textutil.b64encode(textutil.compress(message))
+    except Exception:
+        # If the message could not be encoded and dummy message is returned.
```

Comment:
Typo

If the message could not be encoded, *a* dummy message is returned.

Reply:
Fixed.